src/main/java/mops/businesslogic/PermissionServiceProdImpl.java

```diff
@@ -65,6 +66,21 @@ public String fetchRoleForUserInGroup(Account account, long groupId) throws Mops
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<Group> fetchGroupsForUser(Account account) throws MopsException {
+        Group[] groups = restTemplate.getForObject(URL, Group[].class);
+        if (groups == null) {
+            log.error("The request for groups of user {} failed.", account.getName());
+            throw new GruppenFindungException(String.format(
+                    "Es konnten keinen Gruppen für diese Nutzerin %s gefunden werden.",
+                    account.getName()));
+        }
+        return Arrays.asList(groups);
+    }
+
```

Comment:
This can be in the `GroupService`.

Reply:
Yes that makes sense and life a lot easier.